src/routes/safe/store/actions/transactions/utils/transactionHelpers.ts

```diff
@@ -26,18 +26,35 @@ import { TRANSACTIONS_REDUCER_ID } from 'src/routes/safe/store/reducer/transacti
 import { store } from 'src/store'
 import { safeSelector, safeTransactionsSelector } from 'src/routes/safe/store/selectors'
 import { addOrUpdateTransactions } from 'src/routes/safe/store/actions/transactions/addOrUpdateTransactions'
-import { TxServiceModel } from 'src/routes/safe/store/actions/transactions/fetchTransactions/loadOutgoingTransactions'
+import {
+  BatchProcessTxsProps,
+  TxServiceModel,
+} from 'src/routes/safe/store/actions/transactions/fetchTransactions/loadOutgoingTransactions'
 import { TypedDataUtils } from 'eth-sig-util'
+import { Token } from 'src/logic/tokens/store/model/token'
+import { SafeRecord } from 'src/routes/safe/store/models/safe'
 
 export const isEmptyData = (data?: string | null): boolean => {
   return !data || data === EMPTY_DATA
 }
 
-export const isInnerTransaction = (tx: TxServiceModel, safeAddress: string): boolean => {
-  return sameAddress(tx.to, safeAddress) && Number(tx.value) === 0
+export function isInnerTransaction(tx: Transaction, safeAddress: string): boolean
+export function isInnerTransaction(tx: TxServiceModel, safeAddress: string): boolean
+export function isInnerTransaction(tx: any, safeAddress: string): boolean {
+  let isSameAddress = false
+
+  if (tx.to !== undefined) {
+    isSameAddress = sameAddress(tx.to, safeAddress)
+  } else if (tx.recipient !== undefined) {
+    isSameAddress = sameAddress(tx.recipient, safeAddress)
+  }
+
+  return isSameAddress && Number(tx.value) === 0
 }
 
-export const isCancelTransaction = (tx: TxServiceModel, safeAddress: string): boolean => {
+export function isCancelTransaction(tx: Transaction, safeAddress: string): boolean
+export function isCancelTransaction(tx: TxServiceModel, safeAddress: string): boolean
+export function isCancelTransaction(tx: any, safeAddress: string): boolean {
```

Comment:
What's going on there?

Reply:
function overload (https://www.typescriptlang.org/docs/handbook/functions.html#overloads), as union type didn't worked.

Reply:
Then a different question - why do we have an overload?
Ideally this function should be called only once with either TxServiceModel or Transaction as an argument

Reply:
I've played with it a little bit, this also works 😄 

```
export function isInnerTransaction(tx: TxServiceModel | Transaction, safeAddress: string): boolean {
  let isSameAddress = false

  if ((tx as TxServiceModel).to) {
    isSameAddress = sameAddress((tx as TxServiceModel).to, safeAddress)
  } else if ((tx as Transaction).recipient) {
    isSameAddress = sameAddress((tx as Transaction).recipient, safeAddress)
  }

  return isSameAddress && Number(tx.value) === 0
}

export function isCancelTransaction(tx: TxServiceModel | Transaction, safeAddress: string): boolean {
  return isInnerTransaction(tx, safeAddress) && isEmptyData(tx.data)
}
```

not sure at all if it's better solution though

Reply:
But from what I read in the docs links you shared, function overload is a little bit different thing:
> It’s not uncommon for a single JavaScript function to **return different types of objects based on the shape of the arguments passed in**.

This is about returning different object and not allowing to pass different types of arguments. We're returning a boolean for all cases.

Reply:
Oh... didn't know how to do the `(tx as TxServiceModel)`

I'll update it

Reply:
> Then a different question - why do we have an overload?
> Ideally this function should be called only once with either TxServiceModel or Transaction as an argument

please also pay attention to this comment @fernandomg

Reply:
sorry, I'm not understanding this. The overload was a misuse. And we need to call it using `Transaction` as an argument in here:
https://github.com/gnosis/safe-react/blob/fb03560b5bb1cadc49cbd716ae7e7ca2bc8d9686/src/routes/safe/store/actions/processTransaction.ts#L168-L175

I'm kinda hating this _mock transaction_ approach.